server/models/User.js

```diff
@@ -0,0 +1,46 @@
+const mongoose = require("mongoose");
+const bcrypt = require("bcryptjs");
+const jwt = require("jsonwebtoken");
+
+const UserSchema = new mongoose.Schema({
+  username: {
+    type: String,
+    required: [true, "Please provide a username."],
+    minLength: 6,
+    maxLength: 18,
+    unique: true,
+  },
+  password: {
+    type: String,
+    required: [true, "Please provide a password."],
+    match: [
+      /^(?=.*[0-9])(?=.*[a-z])(?=.*[A-Z])(?=.*[!@$%&?]).{8,32}$/,
+      "Invalid password. Must have at least one lowercase character, one uppercase character, one digit and one special character (!@$%&?).",
+    ],
+    minLength: 8,
+    maxLength: 32,
+  },
+});
+
+UserSchema.pre("save", async function () {
+  const salt = await bcrypt.genSalt(10);
```

Comment:
I would add error handling here - per the bcrypt documentation.

Reply:
Got it.